strawberryfields/api/connection.py

```diff
@@ -301,12 +307,56 @@ def ping(self) -> bool:
             bool: ``True`` if the connection is successful, and ``False`` otherwise
         """
         path = "/healthz"
-        response = requests.get(self._url(path), headers=self._headers)
+        response = self._request("GET", self._url(path), headers=self._headers)
         return response.status_code == 200
 
     def _url(self, path: str) -> str:
         return self._base_url + path
 
+    def _refresh_access_token(self):
+        """Use the offline token to request a new access token."""
+        self._headers.pop("Authorization", None)
```

Comment:
(minor / thinking aloud) I'm always partial to using `del` instead of `pop` (since the latter unnecessarily returns the item as well), but I see that `pop` might be a cleaner solution, since it works even if the item doesn't exist. :laughing: